db_test.go

```diff
@@ -935,6 +935,50 @@ func TestDB_Retention(t *testing.T) {
 	testutil.Equals(t, int64(100), db.blocks[0].meta.MaxTime) // To verify its the right block.
 }
 
+func TestSizeBasedretention(t *testing.T) {
+	db, close := openTestDB(t, &Options{
+		BlockRanges: []int64{100},
+		MaxBytes:    1000,
+	})
+	defer close()
+
+	// Add some empty blocks, each empty block is 249 bytes.
```

Comment:
the only thing it worries me that the test will fail when the default block size changes.

What if you set the `db.Opts.MaxBytes`  after you create the empty block by using it `OpenBlock()` and than `block.Size()`.